module/applications/api/primary-sheet-mixin.mjs

```diff
@@ -0,0 +1,361 @@
+import DragDropApplicationMixin from "../mixins/drag-drop-mixin.mjs";
+
+/**
+ * @import { FilterState5e } from "../components/item-list-controls.mjs";
+ */
+
+/**
+ * Adds V2 sheet functionality shared between primary document sheets (Actors & Items).
+ * @param {typeof DocumentSheet5e} Base  The base class being mixed.
+ * @returns {typeof PrimarySheet5e}
+ */
+export default function PrimarySheetMixin(Base) {
```

Comment:
I wonder if this actually needs to be a mixin. Could it just be the base class for all our document sheets?

```js
export default class PrimaryDocumentSheet extends DragDropApplicationMixin(DocumentSheet5e) {}
```

Reply:
For actors we'll want to mix this into core's `ActorSheetV2` so we get all of the token header actions, rather than the base `DocumentSheetV2`.